File: app/octorise/repo/external/models/ExternalRepo.scala
package octorise.repo.external.models

import javax.persistence._
import org.springframework.beans.factory.annotation.Configurable
import scala.beans.BeanProperty

/**
 * Represents external repository
 */
@Entity
@Table(name="external_repos")
@Inheritance(strategy=InheritanceType.JOINED)
@Configurable
class ExternalRepo {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @BeanProperty
  @Column(name="id")
  var id: Int = _

  @BeanProperty
  @Column(name="name")
  var name: String = _
}

File: app/octorise/repo/external/models/ExternalRepo_.java
package octorise.repo.external.models;

import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

/**
 *
 */
@StaticMetamodel(ExternalRepo.class)
public class ExternalRepo_ {
    public static volatile SingularAttribute<ExternalRepo, Integer> id;
    public static volatile SingularAttribute<ExternalRepo, String> name;
}
